openmetadata-ui/src/main/resources/ui/src/components/NotificationBox/NotificationBox.component.tsx

```diff
@@ -0,0 +1,210 @@
+  onTabChange,
+}: NotificationBoxProp) => {
+  const currentUser = useMemo(
+    () => AppState.getCurrentUserDetails(),
+    [AppState.userDetails, AppState.nonSecureUserDetails]
+  );
+  const [notifications, setNotifications] = useState<Thread[]>([]);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+
+  const [viewAllPath, setViewAllPath] = useState<string>(
+    `${getUserPath(currentUser?.name as string)}/tasks?feedFilter=ASSIGNED_TO`
+  );
+
+  const notificationDropDownList = useMemo(() => {
+    return notifications.slice(0, 5).map((feed, idx) => {
+      const mainFeed = {
+        message: feed.message,
+        postTs: feed.threadTs,
+        from: feed.createdBy,
+        id: feed.id,
+        reactions: feed.reactions,
+      } as Post;
+      const entityType = getEntityType(feed.about);
+      const entityFQN = getEntityFQN(feed.about);
+
+      return (
+        <NotificationFeedCard
+          createdBy={mainFeed.from}
+          entityFQN={entityFQN as string}
+          entityType={entityType as string}
+          feedType={feed.type || ThreadType.Conversation}
+          key={`${mainFeed.from} ${idx}`}
+          taskDetails={feed.task}
+          timestamp={mainFeed.postTs}
+        />
+      );
+    });
+  }, [notifications]);
+
+  const getNotificationData = (
+    threadType: ThreadType,
+    feedFilter: FeedFilter
+  ) => {
+    setIsLoading(true);
+    getFeedsWithFilter(currentUser?.id, feedFilter, undefined, threadType)
+      .then((res: AxiosResponse) => {
+        setNotifications(res.data.data);
+      })
+      .catch((err: AxiosError) => {
+        showErrorToast(
+          err,
+          jsonData['api-error-messages']['fetch-activity-feed-error']
+        );
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
+  };
+
+  const getFilters = (activeTab: ThreadType) => ({
+    threadType: activeTab,
+    feedFilter:
+      activeTab === ThreadType.Task
+        ? FeedFilter.ASSIGNED_TO
+        : FeedFilter.MENTIONS,
+  });
+
+  const updateActiveTab = useCallback(
+    (key: string) => {
+      onTabChange(key);
+      const { threadType, feedFilter } = getFilters(key as ThreadType);
+
+      getNotificationData(threadType, feedFilter);
+
+      setViewAllPath(
+        `${getUserPath(
+          currentUser?.name as string
+        )}/${threadType.toLowerCase()}?feedFilter=${feedFilter}`
+      );
+
+      if (hasTaskNotification || hasMentionNotification) {
+        setTimeout(() => {
+          key === 'Task'
+            ? onMarkTaskNotificationRead()
+            : onMarkMentionsNotificationRead();
+        }, 4000);
+      }
+    },
+    [currentUser, hasTaskNotification, hasMentionNotification]
+  );
+
+  useEffect(() => {
+    getNotificationData(ThreadType.Task, FeedFilter.ASSIGNED_TO);
+  }, []);
+
+  const getTabTitle = (name: string, key: string) => {
+    return (
+      <Badge
+        dot={key === 'Task' ? hasTaskNotification : hasMentionNotification}
```

Comment:
Can we take this value from the function param like how we taking name and key?
```suggestion
        dot={hasNotification}
```